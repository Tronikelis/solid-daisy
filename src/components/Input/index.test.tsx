/* eslint-disable solid/reactivity */
import { render } from "@solidjs/testing-library";
import userEvent from "@testing-library/user-event";
import { createSignal } from "solid-js";
import { describe, expect, it } from "vitest";

import { Input } from ".";

describe("<Input />", () => {
    it("renders correctly", () => {
        const { container } = render(() => <Input />);
        expect(container).toMatchSnapshot();
    });

    it("controlled input works", async () => {
        const [value, setValue] = createSignal("");

        const screen = render(() => (
            <Input
                data-testid="input"
                value={value()}
                onInput={e => setValue(e.target.value)}
            />
        ));

        expect(screen.getByTestId("input")).toHaveValue("");

        setValue("xd");

        expect(screen.getByTestId("input")).toHaveValue("xd");

        await userEvent.type(screen.getByTestId("input"), "foobar");

        expect(value()).toBe("xdfoobar");
        expect(screen.getByTestId("input")).toHaveValue("xdfoobar");
    });

    it("left and right sections render", () => {
        const screen = render(() => (
            <Input leftSection={<div>left</div>} rightSection={<div>right</div>} />
        ));

        expect(screen.getByText("left")).toBeInTheDocument();
        expect(screen.getByText("right")).toBeInTheDocument();
    });

    it("requiredHtml has priority", () => {
        const screen = render(() => (
            <Input data-testid="input" required requiredHtml={false} />
        ));

        expect(screen.getByTestId("input")).not.toBeRequired();
    });
});
