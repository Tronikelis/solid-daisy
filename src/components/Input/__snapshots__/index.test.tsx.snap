// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Input /> > renders correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-0"
  >
    <div
      class="flex flex-col gap-4 justify-center text-inherit input h-auto min-h-12 py-2"
    >
      <div
        class="flex flex-row items-center cursor-[inherit] flex-wrap gap-1"
      >
        <input
          class="pointer-events-auto min-w-6 flex-1 cursor-[inherit] border-none bg-transparent outline-none placeholder-shown:text-ellipsis"
        />
      </div>
    </div>
  </div>
</div>
`;
